### ISSUE TITLE
[BUG]: Opening card browser of a different deck creates issues with reviews

### ISSUE THREAD

**Issue by brishtibheja** [ankidroid/Anki-Android#14143]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

1. Study new card from a selected deck. (I was doing custom study so maybe that is necessary but idk)
2. Open card browser.
3. Click on the deck name above and move to a different deck's card browser
4. Move back to your reviews
5. Continue learning new cards

### Expected behaviour

If I press Again on a new card it should be shown to me after 1 minute.

### Actual behaviour

It doesn't. The learning cards are never shown. When you press back you see that the deck that you were previously viewing on card browser is selected instead of the deck you were studying from.

### Debug info

```text
AnkiDroid Version = 2.16.0beta5

Android Version = 11

ProductFlavor = full

Manufacturer = realme

Model = RMX2193

Hardware = mt6768

Webview User Agent = Mozilla/5.0 (Linux; Android 11; RMX2193 Build/RP1A.200720.011; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/114.0.5735.196 Mobile Safari/537.36

ACRA UUID = dce906d5-19f7-4514-a0e4-0071f4f7b489

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

Something similar also occurred with undo last review where it didn't work at all until I closed and reopened the app. I wasn't able to reproduce that but I believe I was viewing the card browser once again and it was a custom study session(I increased the new card limit for both cases).

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by SanjaySargam** [comment]
Reproduced!

**Comment by yellowjello** [comment]
It sounds like it might share the same root cause as this issue: https://github.com/ankidroid/Anki-Android/issues/13442

ie. changing deck in the card browser -> active deck gets changed -> review queue gets regenerated -> weird things happen due to mismatch in deck state and reviewer state

**Comment by david-allison** [comment]
I believe I've resolved this in 2.17, please let me know if this isn't the case
